core/src/main/java/co/cask/wrangler/statistics/Address.java

```diff
@@ -0,0 +1,196 @@
+  private String unitName = null;
+  private String unitNumber = null;
+  private String city = null;
+  private String state = null;
+  private String zipCode = null;
+
+  public Address() {
+    this.streetNumber = null;
+    this.streetPrefix = null;
+    this.streetName = null;
+    this.streetSuffix = null;
+    this.unitName = null;
+    this.unitNumber = null;
+    this.city = null;
+    this.state = null;
+    this.zipCode = null;
+  }
+
+
+  public Address(String streetNumber, String streetPrefix, String streetName, String streetSuffix,
+                 String unitName, String unitNumber, String city, String state, String zipCode) {
+    this.streetNumber = streetNumber;
+    this.streetPrefix = streetPrefix;
+    this.streetName = streetName;
+    this.streetSuffix = streetSuffix;
+    this.unitName = unitName;
+    this.unitNumber = unitNumber;
+    this.city = city;
+    this.state = state;
+    this.zipCode = zipCode;
+  }
+
+  public String getStreetNumber() {
+    return streetNumber;
+  }
+
+  public String getStreetPrefix() {
+    return streetPrefix;
+  }
+
+  public String getStreetName() {
+    return streetName;
+  }
+
+  public String getStreetSuffix() {
+    return streetSuffix;
+  }
+
+  public String getUnitName() {
+    return unitName;
+  }
+
+  public String getUnitNumber() {
+    return unitNumber;
+  }
+
+  public String getCity() {
+    return city;
+  }
+
+  public String getState() {
+    return state;
+  }
+
+  public String getZipCode() {
+    return zipCode;
+  }
+
+  public void setStreetNumber(String streetNumber) {
+    this.streetNumber = streetNumber;
+  }
+
+  public void setStreetPrefix(String streetPrefix) {
+    this.streetPrefix = streetPrefix;
+  }
+
+  public void setStreetName(String streetName) {
+    this.streetName = streetName;
+  }
+
+  public void setStreetSuffix(String streetSuffix) {
+    this.streetSuffix = streetSuffix;
+  }
+
+  public void setUnitName(String unitName) {
+    this.unitName = unitName;
+  }
+
+  public void setUnitNumber(String unitNumber) {
+    this.unitNumber = unitNumber;
+  }
+
+  public void setCity(String city) {
+    this.city = city;
+  }
+
+  public void setState(String state) {
+    this.state = state;
+  }
+
+  public void setZipCode(String zipCode) {
+    this.zipCode = zipCode;
+  }
+
+  @Override
+  public String toString() {
+    return "Address{" + '\n' +
+            "streetNumber='" + streetNumber + '\'' + '\n' +
+            ", streetPrefix='" + streetPrefix + '\'' + '\n' +
+            ", streetName='" + streetName + '\'' + '\n' +
+            ", streetSuffix='" + streetSuffix + '\'' + '\n' +
+            ", unitName='" + unitName + '\'' + '\n' +
+            ", unitNumber='" + unitNumber + '\'' + '\n' +
+            ", city='" + city + '\'' + '\n' +
+            ", state='" + state + '\'' + '\n' +
+            ", zipCode='" + zipCode + '\'' + '\n' +
+            '}';
+  }
+
+
+  public boolean sameAs(Address other) {
```

Comment:
It will treat {number: 1905, street: Lincoln, suffix: Ave} and {number: null, street: Lincoln, suffix: Ave} as the same address, so I'm not sure if it should be called equals?